python/dgl/graphbolt/item_sampler.py

```diff
@@ -326,8 +329,104 @@ def __iter__(self) -> Iterator:
         data_pipe = data_pipe.collate(collate_fn=self._collate)
 
         # Map to minibatch.
-        data_pipe = data_pipe.map(
-            partial(self._minibatcher, names=self._item_set.names)
+        data_pipe = data_pipe.map(partial(self._minibatcher, names=self._names))
+
+        return iter(data_pipe)
+
+
+@functional_datapipe("distributed_sample_item")
+class DistributedItemSampler(ItemSampler):
+    """Distributed Item Sampler.
+
+    This is useful with torch's DDP training. It will create distributed item
+    subset of data which could be node IDs, node pairs with or without labels,
+    node pairs with negative sources/destinations, DGLGraphs and heterogeneous
+    counterparts. The items of the original item set will be sharded and form
+    an exclusive subset for each replica (process).
+
+    Note: This class `DistributedItemSampler` is not decorated with
+    `torchdata.datapipes.functional_datapipe` on purpose. This indicates it
```

Comment:
I can still see functional_datapipe decoration, conflict?

Reply:
This is the same as ItemSampler (L90). I think it means it's not added **on purpose**.

Reply:
@Rhett-Ying can you clarify the conflict?

Reply:
I have removed the functional decorator in #6357. please remove it too